src/Search/Processors/SearchUpdateProcessor.php

```diff
@@ -104,6 +111,10 @@ protected function prepareIndexes()
                     }
                 }
             }
+
+            foreach (array_keys($indexesThatDeleted) as $index) {
+                $indexes[$index]->commit();
```

Comment:
Should use the highlevel API provided here - `commitIndex()`.

So in case an index is both "dirty" and had deletions (which is quite likely), commit is now called twice - once in `prepareIndexes()`, and once in the callee of that (`process()`). I assume since there's nothing to commit the second time around, that's a pretty cheap operation. But it feels hacky. For starters, if the new commits fail, the parent method won't return false. `prepareIndexes()` is a protected method, so technically it's an API change to return *both* index sets, but I think it's consistent with the meaning of the method, so would advocate to just use the existing `commit()` invocation.

Reply:
- Should use the highlevel API provided here - commitIndex()
- so would advocate to just use the existing commit() invocation.

I'm a little unsure what you're asking me to do here, could you please clarify

Reply:
Hah sorry, I can see how that's confusing. The important bit here is that there's only one commit per index for *both* indexes with dirty states and deletes. `commitIndex()` already gets called in `process()`. You shouldn't need to call `commit()` separately yourself, you should adjust the `prepareIndexes()` method return value instead to incorporate the additional indexes.

Reply:
ok, have removed the double commit. have simply ensured that `$dirtyIndexes[$index] = $indexes[$index];` is run on both add + delete